src/core/qgscoordinatetransform.cpp

```diff
@@ -637,6 +637,13 @@ void QgsCoordinateTransform::transformCoords( int numPoints, double *x, double *
     return;
   }
 
+  double *xprev = new double[ numPoints ];
+  memcpy( xprev, x, sizeof( double ) * numPoints );
+  double *yprev = new double[ numPoints ];
+  memcpy( yprev, y, sizeof( double ) * numPoints );
+  double *zprev = new double[ numPoints ];
+  memcpy( zprev, z, sizeof( double ) * numPoints );
```

Comment:
this method cannot be called with z==NULL right ?

Reply:
right -- all code paths leading here pad out the z array with 0's if it doesn't already exist